sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/SessionReceiverManager.cs

```diff
@@ -177,7 +183,7 @@ private async Task CreateReceiver(CancellationToken processorCancellationToken)
                     return;
                 }
                 _threadCount--;
-                if (_threadCount == 0)
+                if (_threadCount == 0 && !processorCancellationToken.IsCancellationRequested)
```

Comment:
Does this need to be thread safe?  You may want to move it to an interlocked call.

Reply:
This is within a lock.